front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PluginArtifactController.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.controllers;
+
+import com.netflix.spinnaker.front50.model.pluginartifact.PluginArtifact;
+import com.netflix.spinnaker.front50.model.pluginartifact.PluginArtifactService;
+import java.util.Collection;
+import java.util.Optional;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/** TODO(rz): What's the permissions model for something like artifacts? */
+@RestController
+@RequestMapping("/pluginArtifacts")
+public class PluginArtifactController {
+
+  private final PluginArtifactService pluginArtifactService;
+
+  public PluginArtifactController(PluginArtifactService pluginArtifactService) {
+    this.pluginArtifactService = pluginArtifactService;
+  }
+
+  @RequestMapping(value = "", method = RequestMethod.GET)
+  Collection<PluginArtifact> list(
+      @RequestParam(value = "service", required = false) String service) {
+    return Optional.ofNullable(service)
+        .map(pluginArtifactService::findAllByService)
+        .orElseGet(pluginArtifactService::findAll);
+  }
+
+  @RequestMapping(value = "", method = RequestMethod.POST)
+  PluginArtifact create(@RequestBody PluginArtifact pluginArtifact) {
+    return pluginArtifactService.upsert(pluginArtifact);
+  }
+
+  @RequestMapping(value = "/{id}", method = RequestMethod.POST)
```

Comment:
The `id` is not passed to `upsert`, and even if it were passed to `upsert` -- if the `id` doesn't exist the upsert creates the plugin anyways.  

Do we want to provide a pattern for passing a known plugin ID and plugin artifact?  I'm not really sure I understand the point of this endpoint.

Reply:
Ha. You're right. I'm going to keep the endpoint (since it follows convention), but just assert they're the same.

Reply:
Jk, that's not what I'm doing.